packages/edit-site/src/hooks/use-template-info.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { find } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Given a template entity, return information about it which is ready to be
+ * rendered, such as the title and description.
+ *
+ * @param {Object} template The template for which we need information.
+ * @return {Object} Information about the template, including title and description.
+ */
+export default function useTemplateInfo( template ) {
+	const defaultTemplateTypesDefinitions = useSelect(
+		( select ) =>
+			select( 'core/edit-site' ).getDefaultTemplateTypesDefinitions(),
+		[]
+	);
+
+	if ( ! template ) {
+		return {};
+	}
+
+	if ( 'wp_template_part' === template.type ) {
```

Comment:
I might be missing something, but as far as I see, `getDefaultTemplateTypesDefinitions` returns the settings object exposed from the server where should be only `slug`, `title` and `description`. Can you confirm that `type` is injected there as well?

Reply:
I think this is probably because the template CPT object would be passed here, which probably has `type` set to `postType` automatically at some point in the API middleware?

Reply:
As @noahtallen said. 🙂 

The `template` parameter is a `wp_template` (or `wp_template_part`) entity, so it does have the `type` property.

Basically the hook takes the template (or template part) entity stored in Redux, and returns its title (and description) in this order of preference:
- The actual entity title if provided.
- The default title if exists in the default definitions list.
- The slug.